clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -3562,6 +3562,28 @@ def LLVMIntrinsicCallOp : CIR_Op<"llvm.intrinsic"> {
 
 }
 
+//===----------------------------------------------------------------------===//
+// InvariantGroupOp
+//===----------------------------------------------------------------------===//
+
+def InvariantGroupOp
+    : CIR_Op<"invariant_group", [Pure, SameOperandsAndResultType]> {
+  let summary = "Start an invariant group";
+  let description = [{
+    The `cir.invariant_group` operation takes a single pointer value as argument
+    and returns the same pointer value with a fresh invariant group. All loads
+    and stores that access the returned pointer value are presumed by the
+    optimizer to load or store the same value.
```

Comment:
Perhaps add a bit of info (link) to what invariant group means? Also another paragraph mentioning this operation is used before lowering (not CIRGen) and what it aims at achieving

Reply:
Updated.